data_safe_haven/commands/shm.py

```diff
@@ -114,4 +116,5 @@ def teardown() -> None:
         pulumi_config.upload(context)
     except DataSafeHavenError as exc:
         msg = "Could not teardown Safe Haven Management environment."
-        raise DataSafeHavenError(msg) from exc
+        logger.critical(msg)
```

Comment:
Could probably put the little string straight in the `logger.critical()` call here.

The reason we often have

```python
msg = "..."
raise Exception(message)
```

is for linting.